internal/controllers/token_exchange_test.go

```diff
@@ -50,13 +51,23 @@ func TestTokenExchangeController_GetDeviceCommandPermissionWithScope(t *testing.
 	client := ethclient.Client{}
 
 	// setup app and route req
-	c := NewTokenExchangeController(&logger, &config.Settings{
+	c, err := NewTokenExchangeController(&logger, &config.Settings{
 		BlockchainNodeURL:        "http://testurl.com/mock",
 		ContractAddressWhitelist: "",
 		ContractAddressSacd:      "0xa6",
 	}, dexService, usersSvc, contractsMgr, &client)
+	if err != nil {
+		require.NoError(t, err, "Failed to initialize token exchange controller")
+	}
```

Comment:
nit: you don't need the nil check either but this works either way
```suggestion
	require.NoError(t, err, "Failed to initialize token exchange controller")
```